[RELEASE] Release version 2.12.0

This is a component issue for `2.12.0`.
Coming from https://github.com/opensearch-project/opensearch-build/issues/4115. Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help.
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `2.x` branch to `2.12.0` for Min/Core, and `2.12.0.0` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [ ] Assign this issue to a release owner.
- [ ] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [ ] All the tasks in this issue have been reviewed by the release owner.
- [ ] Create, update, triage and label all features and issues targeted for this release with `v2.12.0`.
- [ ] Finalize the code and create the the release branch `2.12` from the `2.x` branch.

### CI/CD

- [ ] All code changes for `2.12.0` are complete.
- [ ] Ensure working and passing CI.
- [ ] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.12.0).

### Pre-Release

- [ ] Increment the version on the parent branch to the next development iteration.
- [ ] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [ ] Confirm that all changes for `2.12.0` have been merged.
- [ ] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) for the next developer iteration.

### Release Testing

- [ ] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [ ] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [ ] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [ ] File issues for all intermittent test failures.

### Release

- [ ] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [ ] Verify all issued labeled for this release are closed or labelled for the next release.
- [ ] Verify the release date mentioned in release notes is correct and matches actual release date.

### Post Release

- [ ] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [ ] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [ ] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/component_release_template.md).
- [ ] Conduct a retrospective, and publish its results.

Please check this error, OSD 2.12.0 failed to start due to 
https://github.com/opensearch-project/opensearch-build/issues/4115#issuecomment-1942177293
```
opensearch-dashboards  | {"type":"log","@timestamp":"2024-02-13T19:54:26Z","tags":["fatal","root"],"pid":1,"message":"Error: Cannot find module '../config'\nRequire stack:\n- /usr/share/opensearch-dashboards/plugins/securityAnalyticsDashboards/server/index.js\n- /usr/share/opensearch-dashboards/src/core/server/plugins/plugin.js\n- /usr/share/opensearch-dashboards/src/core/server/plugins/discovery/plugins_discovery.js\n- /usr/share/opensearch-dashboards/src/core/server/plugins/discovery/index.js\n- /usr/share/opensearch-dashboards/src/core/server/plugins/plugins_service.js\n- /usr/share/opensearch-dashboards/src/core/server/plugins/index.js\n- /usr/share/opensearch-dashboards/src/core/server/index.js\n- /usr/share/opensearch-dashboards/src/cli/serve/serve.js\n- /usr/share/opensearch-dashboards/src/cli/cli.js\n- /usr/share/opensearch-dashboards/src/cli/dist.js\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)\n    at Module.Hook._require.Module.require (/usr/share/opensearch-dashboards/node_modules/require-in-the-middle/index.js:61:29)\n    at require (node:internal/modules/helpers:177:18)\n    at Object.<anonymous> (/usr/share/opensearch-dashboards/plugins/securityAnalyticsDashboards/server/index.ts:9:1)\n    at Module._compile (node:internal/modules/cjs/loader:1356:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1414:10)\n    at Module.load (node:internal/modules/cjs/loader:1197:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:1013:12)\n    at Module.require (node:internal/modules/cjs/loader:1225:19)\n    at Module.Hook._require.Module.require (/usr/share/opensearch-dashboards/node_modules/require-in-the-middle/index.js:80:39) {\n  code: 'MODULE_NOT_FOUND',\n  requireStack: [\n    '/usr/share/opensearch-dashboards/plugins/securityAnalyticsDashboards/server/index.js',\n    '/usr/share/opensearch-dashboards/src/core/server/plugins/plugin.js',\n    '/usr/share/opensearch-dashboards/src/core/server/plugins/discovery/plugins_discovery.js',\n    '/usr/share/opensearch-dashboards/src/core/server/plugins/discovery/index.js',\n    '/usr/share/opensearch-dashboards/src/core/server/plugins/plugins_service.js',\n    '/usr/share/opensearch-dashboards/src/core/server/plugins/index.js',\n    '/usr/share/opensearch-dashboards/src/core/server/index.js',\n    '/usr/share/opensearch-dashboards/src/cli/serve/serve.js',\n    '/usr/share/opensearch-dashboards/src/cli/cli.js',\n    '/usr/share/opensearch-dashboards/src/cli/dist.js'\n  ]\n}"}
```

--- a/.opensearch_dashboards-plugin-helpers.json
+++ b/.opensearch_dashboards-plugin-helpers.json
@@ -5,6 +5,7 @@
     "yarn.lock",
     ".yarnrc",
     "{lib,public,server,webpackShims,translations,utils,models,test,common,types}/**/*",
-    "!__tests__"
+    "!__tests__",
+    "config.ts"
   ]
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,7 +3,16 @@
   "extends": "../../tsconfig.json",
 
   // tell the TypeScript compiler where to find your source files
-  "include": ["server/**/*", "public/**/*", "utils/**/*", "models/**/*", "test/**/*", "common/**/*", "types/**/*"],
+  "include": [
+    "server/**/*",
+    "public/**/*",
+    "utils/**/*",
+    "models/**/*",
+    "test/**/*",
+    "common/**/*",
+    "types/**/*",
+    "config.ts"
+  ],
   "exclude": ["node_modules", "*/node_modules/"],
   "compilerOptions": {
     "skipLibCheck": true,

